src/otx/algo/detection/backbones/presnet.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Presnet backbones, modified from https://github.com/lyuwenyu/RT-DETR."""
+
+from __future__ import annotations
+
+from collections import OrderedDict
+from typing import Any
+
+import torch
+from torch import nn
+
+from otx.algo.modules import ConvModule, build_activation_layer
+
+__all__ = ["PResNet"]
+
+
+ResNet_cfg = {
+    18: [2, 2, 2, 2],
+    34: [3, 4, 6, 3],
+    50: [3, 4, 6, 3],
+    101: [3, 4, 23, 3],
+}
```

Comment:
What configuration of ResNet does this mean exactly?

Reply:
Number of blocks for each stage. I will rename it and move to the class

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/3741/commits/2d5cef1ef2bbc576bebd10fd1a96e812d6ef0548